Mage.Sets/src/mage/cards/t/TheIrencrag.java

```diff
@@ -0,0 +1,128 @@
+package mage.cards.t;
+
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldAllTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.abilities.effects.common.AddContinuousEffectToGame;
+import mage.abilities.effects.common.continuous.BoostEquippedEffect;
+import mage.abilities.keyword.EquipAbility;
+import mage.abilities.mana.ColorlessManaAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterControlledCreaturePermanent;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+
+import java.util.UUID;
+
+/**
+ * @author Susucr
+ */
+public final class TheIrencrag extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterControlledCreaturePermanent("a legendary creature");
+
+    static {
+        filter.add(SuperType.LEGENDARY.getPredicate());
+        filter.add(TargetController.YOU.getControllerPredicate());
+    }
+
+    public TheIrencrag(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{2}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+
+        // {T}: Add {C}.
+        this.addAbility(new ColorlessManaAbility());
+
+        // Whenever a legendary creature enters the battlefield under your control, you may have The Irencrag become a legendary Equipment artifact named Everflame, Heroes' Legacy. If you do, it gains equip {3} and "Equipped creature gets +3/+3" and loses all other abilities.
+        this.addAbility(new EntersBattlefieldAllTriggeredAbility(
```

Comment:
`EntersBattlefieldControlledTriggeredAbility`?

(I have no idea why `EntersBattlefieldAllTriggeredAbility` has that `controlledText` parameter as a red herring)

Reply:
it's from before the `setTriggerPhrase` refactor